src/Microsoft.ServiceFabric.Services.Remoting/Client/ServiceProxyFactory.cs

```diff
@@ -91,65 +91,45 @@ public ServiceProxyFactory(OperationRetrySettings retrySettings = null)
         {
             var serviceInterfaceType = typeof(TServiceInterface);
 
-#if !DotNetCoreClr
-
             //Use provider to find the stack
-            if (this.proxyFactoryV1 == null && this.proxyFactoryV2 == null)
+            var provider = this.GetProviderAttribute(serviceInterfaceType);
+
+#if !DotNetCoreClr
+            if (provider.RemotingClient.Equals(RemotingClient.V1Client))
             {
-                var provider = this.GetProviderAttribute(serviceInterfaceType);
-                if (provider.RemotingClient.Equals(RemotingClient.V2Client))
-                {
-                    //We are overriding listenerName since using provider we can have multiple listener configured(Compat Mode).
-                    this.overrideListenerName = true;
-                    this.proxyFactoryV2 =
-                        new V2.Client.ServiceProxyFactory(provider.CreateServiceRemotingClientFactoryV2,this.retrySettings);
-                }
-                else
+                // Create V1 proxyFactory
+                if (this.proxyFactoryV1 == null)
                 {
-                    this.proxyFactoryV1 = new V1.Client.ServiceProxyFactory(provider.CreateServiceRemotingClientFactory,this.retrySettings);
+                    this.proxyFactoryV1 =
+                        new V1.Client.ServiceProxyFactory(provider.CreateServiceRemotingClientFactory, this.retrySettings);
                 }
-                
-           }
 
-            if (this.proxyFactoryV1 != null)
-            {
                 return this.proxyFactoryV1.CreateServiceProxy<TServiceInterface>(serviceUri,
                     partitionKey,
                     targetReplicaSelector,
                     listenerName);
             }
-            if (this.overrideListenerName && listenerName==null)
-            {
-                return this.proxyFactoryV2.CreateServiceProxy<TServiceInterface>(serviceUri,
-                    partitionKey,
-                    targetReplicaSelector,
-                    ServiceRemotingProviderAttribute.DefaultV2listenerName);
-            }
-            return this.proxyFactoryV2.CreateServiceProxy<TServiceInterface>(serviceUri,
-                partitionKey,
-                targetReplicaSelector,
-                listenerName);
-#else
-            if (proxyFactoryV2==null)
+#endif
+
+            // Create V2 proxyFactory
+            if (this.proxyFactoryV2 == null)
             {
-                var provider = this.GetProviderAttribute(serviceInterfaceType);
-                //We are overriding listenerName since using provider we can have multiple listener configured(Compat Mode).
-                this.overrideListenerName = true;
```

Comment:
What is `overrideListenerName` for? Looks like it was just used to determine if we were using V2 listener over V1.

It is not needed in my code so perhaps remove defn for it.

Reply:
This is to figure out wether to override the listenerName is a named listener is provided.

Reply:
Ah, I see, it's so you can figure out if an implementaion has been given or not in the constructor. IF not, then it uses a default V2 implementation which requires a listener name of `ServiceRemotingProviderAttribute.DefaultV2listenerName`